services/graph/pkg/service/v0/educationschools.go

```diff
@@ -195,17 +195,28 @@ func (g Graph) DeleteEducationSchool(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	logger.Debug().Str("schoolID", schoolID).Msg("Getting users of school")
+	users, err := g.identityEducationBackend.GetEducationSchoolUsers(r.Context(), schoolID)
+	if err != nil {
+		logger.Debug().Err(err).Msg("could not get school users: backend error")
+		renderInternalServerError(w, r, err)
+		return
+	}
+
+	for _, user := range users {
+		logger.Debug().Str("schoolID", schoolID).Str("userID", *user.Id).Msg("calling delete member on backend")
+		if err := g.identityEducationBackend.RemoveUserFromEducationSchool(r.Context(), schoolID, *user.Id); err != nil {
```

Comment:
I wonder what's the best error handling behavior here. We might not want to refuse to deleting a school if removing a single member fails (might e.g. just be that the user to deleted by something else between the GetEducationSchoolUsers() call and here).

Maybe it's better to just log a warning here and continue with the deletion process.

@butonic thoughts?

Reply:
I was also in doubt on that, I went with this because I considered having uncleaned up users to be a worse state than erroring out if it didn't work.

TBH, it would be nice to make this an atomic operation.

Reply:
@rhafer could we not make this atomic by storing a last modified attribute and sending an additional filter with the last modified date?

Reply:
No. I don't think that something like this is possible

Reply:
Yeah I will check for `errNotFound`, it /should/ never be hit as it gets the users from the backend right before that, but we all know how those kinds of assumptions tend to explode in our faces. :)

Reply:
One problem that just occured to me is that `errNotFound` is not a public error and resides in the identity package. So I can't use `errors.Is` for it. I could match on the string, but that's not really a clean way of doing that.

Reply:
After making `errNotFound` public, I now continue on `ErrNotFound`.